File: java/src/main/java/cn/huidu/device/api/sdk/data/program/other/ClockParam.java
package cn.huidu.device.api.sdk.data.program.other;

public class ClockParam {
    
    public class FormatColor {
        public String display;
        public String format;
        public String color;
        public int size;
    }
    
    public class StringColor {
        public String display;
        public String string;
        public int size;
        public String color;
    }
    
    public class Style {
        public String hourHandColor;
        public String minuteHandColor;
        public String secondHandColor;
        public String hourScaleColor;
        public String minuteScaleColor;
        public String fontName;
        public int fontSize;
    }

    public class FmtStyle {
        public String swap;
		public String space;
	}
}

File: python/sdk/data/other/ClockParam.py
class ClockParam:
    class FormatColor:
        def __init__(self, display: str = None, format: str = None,
                     color: str = None, size: int = None):
            self.display = display  # Text to display
            self.format = format    # Format pattern (e.g., "HH:mm:ss")
            self.color = color      # Text color value
            self.size = size        # Font size

    class StringColor:
        def __init__(self, display: str = None, string: str = None,
                     size: int = None, color: str = None):
            self.display = display  # Display label
            self.string = string    # Actual string content
            self.size = size         # Font size
            self.color = color       # Text color value

    class Style:
        def __init__(self, hourHandColor: str = None, minuteHandColor: str = None,
                     secondHandColor: str = None, hourScaleColor: str = None,
                     minuteScaleColor: str = None, fontName: str = None,
                     fontSize: int = None):
            self.hourHandColor = hourHandColor    # Hour hand color
            self.minuteHandColor = minuteHandColor  # Minute hand color
            self.secondHandColor = secondHandColor  # Second hand color
            self.hourScaleColor = hourScaleColor    # Hour scale/mark color
            self.minuteScaleColor = minuteScaleColor  # Minute scale/mark color
            self.fontName = fontName              # Font family name
            self.fontSize = fontSize              # Base font size

    class FmtStyle:
        def __init__(self, swap: str = None, space: str = None):
            self.swap = swap    # AM/PM swap indicator
            self.space = space  # Space formatting character